File: src/metrics.py
class MetricsCollector:
    def __init__(self, total: int):
        self.killed = 0
        self.resolved = 0
        self.false_positives = 0
        self.completed = 0
        self.total = total

    def throughput(self) -> float:
        if self.total == 0:
            return 0.0
        return self.completed / self.total

    def false_positive_rate(self) -> float:
        denom = self.resolved + self.false_positives
        if denom == 0:
            return 0.0
        return self.false_positives / denom

    def summary(self) -> dict:
        return {
            "killed": self.killed,
            "resolved": self.resolved,
            "false_positives": self.false_positives,
            "completed": self.completed,
            "total": self.total,
            "throughput": self.throughput(),
            "false_positive_rate": self.false_positive_rate(),
        }

File: src/metrics.c
#include "metrics.h"
#include <string.h>

void metrics_init(Metrics* m, int total) {
    memset(m, 0, sizeof(Metrics));
    m->total = total;
}

double metrics_throughput(const Metrics* m) {
    if (m->total == 0) return 0.0;
    return (double)m->completed / (double)m->total;
}

double metrics_fp_rate(const Metrics* m) {
    int denom = m->resolved + m->false_positives;
    if (denom == 0) return 0.0;
    return (double)m->false_positives / (double)denom;
}
